packages/ai-chat/src/browser/session-summary-variable-contribution.ts

```diff
@@ -0,0 +1,115 @@
+        service.registerArgumentPicker(SESSION_SUMMARY_VARIABLE, this.pickSession.bind(this));
+        service.registerArgumentCompletionProvider(SESSION_SUMMARY_VARIABLE, this.provideCompletionItems.bind(this));
+    }
+
+    protected async pickSession(): Promise<string | undefined> {
+        const items = this.getItems();
+        const selection = await this.quickInputService.showQuickPick(items);
+        return selection?.id;
+    }
+
+    protected async provideCompletionItems(
+        model: monaco.editor.ITextModel,
+        position: monaco.Position,
+        matchString?: string
+    ): Promise<monaco.languages.CompletionItem[] | undefined> {
+        const context = AIVariableCompletionContext.get(SESSION_SUMMARY_VARIABLE.name, model, position, matchString);
+        if (!context) { return undefined; }
+        const { userInput, range, prefix } = context;
+        return this.getItems().filter(candidate => QuickPickItem.is(candidate) && candidate.label.startsWith(userInput)).map(({ label, id }: QuickPickItem) => ({
+            label,
+            kind: monaco.languages.CompletionItemKind.Class,
+            range,
+            insertText: `${prefix}${id}`,
+            detail: id,
+            filterText: userInput,
+        }));
+    }
+
+    protected getItems(): QuickPickItemOrSeparator[] {
+        return [
+            ...(this.summaries.size ? [{ type: 'separator', label: 'Recent Summaries' }] satisfies QuickPickSeparator[] : []),
+            ...Array.from(this.summaries.entries(), ([id, { label }]) => ({
+                type: 'item',
+                label,
+                id
+            })) satisfies QuickPickItem[],
+            ...(this.summaries.size ? [{ type: 'separator', label: 'Other Sessions' }] satisfies QuickPickSeparator[] : []),
+            ...this.chatService.getSessions()
+                .filter(candidate => !this.summaries.has(candidate.id) && candidate.model.getRequests().length)
+                .map<QuickPickItem>(session => ({ type: 'item', label: session.title || session.id, id: session.id }))
+        ];
+    }
+
+    canResolve(request: AIVariableResolutionRequest, context: AIVariableContext): MaybePromise<number> {
+        return request.variable.id === SESSION_SUMMARY_VARIABLE.id ? 10000 : -5;
+    }
+
+    async resolve(request: AIVariableResolutionRequest, context: AIVariableContext): Promise<ResolvedAIContextVariable | undefined> {
+        if (request.variable.id !== SESSION_SUMMARY_VARIABLE.id || !request.arg) { return; }
+        const existingSession = this.chatService.getSession(request.arg);
+        const existingSummary = this.summaries.get(request.arg);
+        const newSummaryPossibleAndNecessary = !!existingSession && (!existingSummary || existingSummary.length !== existingSession.model.getRequests().length);
+        try {
+            const value = newSummaryPossibleAndNecessary ? await this.summarizeSession(existingSession) : existingSummary?.summary;
```

Comment:
This would call the TaskContextService instead

Reply:
Done.